src/components/pages/ProblemIndex/index.test.tsx

```diff
@@ -1,17 +1,138 @@
-import { screen, fireEvent } from "@testing-library/react";
+import { screen, fireEvent, within } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
 import { testRenderer } from "../../../setupTests";
 
 import ProblemIndex from "./index";
 
 describe("<ProblemIndex/>", () => {
+  it("displays inputted value", async () => {
+    testRenderer(<ProblemIndex />);
+
+    const input = (await screen.findByLabelText(
+      "Szukaj problemów"
+    )) as HTMLInputElement;
+    expect(input.value).toBe("");
+
+    fireEvent.change(input, { target: { value: "Good Day" } });
+
+    expect(input.value).toBe("Good Day");
+  });
+
+  it("should show suggestions list matching input value", async () => {
+    testRenderer(<ProblemIndex />);
+
+    const input = (await screen.findByLabelText(
+      "Szukaj problemów"
+    )) as HTMLInputElement;
+    expect(input.value).toBe("");
+    expect(screen.queryByRole("listbox")).not.toBeInTheDocument();
+
+    fireEvent.change(input, { target: { value: "c" } });
```

Comment:
Do we want in every place like this to use userEvent instead of fireEvent? Or just in some cases?

Reply:
Everywhere, that's a good catch! :) I will fix it everywhere.
It's better to use userEvent package as it is closer to actual user experience, here it's a nice and short [explanation by Kent C. Dodds](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-testing-libraryuser-event)